x/evm/types/utils.go

```diff
@@ -38,11 +38,11 @@ func ValidateSigner(signBytes, sig []byte, signer ethcmn.Address) error {
 }
 
 func rlpHash(x interface{}) (hash ethcmn.Hash) {
-	hasher := ethsha.NewKeccak256()
+	hasher := sha3.NewLegacyKeccak256()
```

Comment:
should probably be using geth? https://github.com/ethereum/go-ethereum/blob/391d4cb9b53eb9f71862cbb8a6cca332d5a76f94/crypto/crypto.go#L45

Reply:
yeah, I had changed it previously to get around the replace issues, but it should be good to know what the difference in these functions are because it didn't affect anything here to change it, either because it is the same in our usable cases or it isn't currently used. Will revert now tho good catch

Reply:
I actually do remember what the issue was, this function is replacing a private function that is not accessible inside ethermint, using the function you linked doesn't work. This is the function that is being replaced: https://github.com/ethereum/go-ethereum/blob/391d4cb9b53eb9f71862cbb8a6cca332d5a76f94/core/types/block.go#L111